src/react/SessionManager.ts

```diff
@@ -0,0 +1,267 @@
+/*
+   This file is part of Astarte.
+
+   Copyright 2020 Ispirata Srl
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+import { createNanoEvents, Emitter, Unsubscribe } from 'nanoevents';
+
+type Storage = {
```

Comment:
I know why you want to add this here, but it's out of scope for this PR, let's take it one step at the time

Reply:
Are you referring to localStorage or nanoevents?

Reply:
I removed the Storage type and avoided exposing a `storage` param for SessionManager's initialization.
I just kept an internal reference to localStorage as a cleaner approach.